Signal/src/ViewControllers/FullImageViewController.h

```diff
@@ -2,21 +2,18 @@
 //  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
 //
 
-#import "OWSMessageData.h"
 #import "OWSViewController.h"
-#import "TSAttachmentStream.h"
-#import "TSInteraction.h"
-#import <UIKit/UIKit.h>
 
 NS_ASSUME_NONNULL_BEGIN
 
+@class ConversationViewItem;
+@class TSAttachmentStream;
+
 @interface FullImageViewController : OWSViewController
 
-- (instancetype)initWithAttachment:(TSAttachmentStream *)attachment
+- (instancetype)initWithAttachment:(TSAttachmentStream *)attachmentStream
                           fromRect:(CGRect)rect
-                    forInteraction:(TSInteraction *)interaction
-                       messageItem:(id<OWSMessageData>)messageItem
-                        isAnimated:(BOOL)animated;
+                          viewItem:(ConversationViewItem *)viewItem;
```

Comment:
By passing in the view item, we can DRY up a bunch of logic, making this class simpler and more consistent.